contracts/BalanceTrackerFixedPrice.sol

```diff
@@ -0,0 +1,332 @@
+        if (actualDeliveryRate == 0) {
+            revert ZeroValue();
+        }
+
+        // Get token associated with request Id
+        address token = mapRequestIdTokens[requestId];
+
+        uint256 rateDiff;
+        if (maxDeliveryRate > actualDeliveryRate) {
+            // Return back requester overpayment debit
+            rateDiff = maxDeliveryRate - actualDeliveryRate;
+            mapRequesterBalances[requester][token] += rateDiff;
+        } else {
+            actualDeliveryRate = maxDeliveryRate;
+        }
+
+        // Record payment into mech balance
+        mapMechBalances[mech][token] += actualDeliveryRate;
+
+        emit MechPaymentCalculated(mech, requestId, actualDeliveryRate, rateDiff);
+    }
+
+    // TODO buyBackBurner does not account for other tokens but WETH, OLAS
+    /// @dev Drains collected fees by sending them to a Buy back burner contract.
+    function drain(address token) external {
+        // Reentrancy guard
+        if (_locked > 1) {
+            revert ReentrancyGuard();
+        }
+        _locked = 2;
+
+        uint256 localCollectedFees = mapCollectedFees[token];
+
+        // TODO Limits
+        // Check for zero value
+        if (localCollectedFees == 0) {
+            revert ZeroValue();
+        }
+
+        mapCollectedFees[token] = 0;
+
+        // Check token address
+        if (token == address (0)) {
+            // Wrap native tokens
+            _wrap(localCollectedFees);
+            // Transfer to Buy back burner
+            IToken(wrappedNativeToken).transfer(buyBackBurner, localCollectedFees);
+        } else {
+            IToken(token).transfer(buyBackBurner, localCollectedFees);
+        }
+
+        emit Drained(token, localCollectedFees);
+
+        _locked = 1;
+    }
+
+    function _withdraw(address token, uint256 balance) internal {
+        bool success;
+        // Transfer mech balance
+        if (token == address(0)) {
+            // solhint-disable-next-line avoid-low-level-calls
+            (success, ) = msg.sender.call{value: balance}("");
+        } else {
+            IToken(token).transfer(msg.sender, balance);
+        }
+
+        // Check transfer
+        if (!success) {
+            revert TransferFailed(token, address(this), msg.sender, balance);
+        }
+    }
+
+    /// @dev Processes mech payment by withdrawing funds.
+    function processPayment(address token) external returns (uint256 mechPayment, uint256 marketplaceFee) {
+        // Reentrancy guard
+        if (_locked > 1) {
+            revert ReentrancyGuard();
+        }
+        _locked = 2;
+
+        // Get mech balance
+        uint256 balance = mapMechBalances[msg.sender][token];
+        // TODO limits?
+        if (balance == 0) {
+            revert ZeroValue();
+        }
+
+        // Calculate mech payment and marketplace fee
+        uint256 fee = IMechMarketplace(mechMarketplace).fee();
+        marketplaceFee = (balance * fee) / 10_000;
```

Comment:
See above, suggest rounding up so we always incentivise later, rather than earlier withdrawals